app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsAdapter.kt

```diff
@@ -103,19 +107,25 @@ class PatientSearchResultsAdapter @Inject constructor(
         phoneNumberTextView.text = phoneObfuscator.obfuscate(phoneNumber!!)
       }
 
-      // TODO: Join BP table with patient search query.
-      val lastBpDate: LocalDate? = null
-      val lastBpFacilityName: String? = null
+      val lastBp = searchResult.lastBp
+      if (lastBp == null) {
+        lastBpDateFrame.visibility = View.GONE
 
-      if (lastBpDate == null) {
-        lastBpDateTextView.visibility = View.GONE
       } else {
-        lastBpDateTextView.visibility = View.VISIBLE
+        lastBpDateFrame.visibility = View.VISIBLE
+
+        val lastBpDate = lastBp.takenOn.atZone(UTC).toLocalDate()
         val formattedLastBpDate = DATE_OF_BIRTH_FORMATTER.format(lastBpDate)
-        lastBpDateTextView.text = resources.getString(
```

Comment:
Should we rename `lastBpDateTextView`  to `lastBpDateAndFacilityTextView`?

Reply:
Done. Renamed this to `lastBpDateAndFacilityTextView`